### PR TITLE
Upgrade to storybook v8.0.5

### PR DESCRIPTION
**Pull request by SiAdcock**
## What are you changing?

- upgrade storybook to v8.0.5
- remove [deprecated `@storybook/addons`](https://www.npmjs.com/package/@storybook/addons)

## Why?

I naively thought an upgrade might fix the [accessibility scan](https://github.com/guardian/csnx/issues/1121#issuecomment-2026186476) issue. Alas it did not. However I took the opportunity to remove a deprecated package whilst I was here, which was causing version matching issues.

### CODE REVIEW

File: libs/@guardian/source-react-components-development-kitchen/.storybook/preview.tsx [hunk 1]
```diff
@@ -20,4 +21,11 @@ export const parameters = {
 	},
 };
 
+const preview: Preview = {
+	parameters,
+};
+
+/* eslint-disable-next-line import/no-default-export -- this is the preview api */
+export default preview;
```
Comment (SiAdcock) [inline libs/@guardian/source-react-components-development-kitchen/.storybook/preview.tsx:29]: See https://storybook.js.org/docs/essentials/backgrounds#configuration

File: package.json [hunk 2]
```diff
@@ -24,15 +24,15 @@
 		"@nx/eslint-plugin": "18.0.7",
 		"@nx/jest": "18.0.7",
 		"@nx/workspace": "18.0.7",
-		"@storybook/addon-a11y": "8.0.0",
-		"@storybook/addon-docs": "8.0.0",
-		"@storybook/addon-essentials": "8.0.0",
-		"@storybook/addon-links": "8.0.0",
-		"@storybook/addon-viewport": "8.0.0",
-		"@storybook/addons": "7.6.5",
```
Comment (SiAdcock) [inline package.json:32]: This package is [deprecated](https://www.npmjs.com/package/@storybook/addons)

File: package.json [hunk 3]
```diff
@@ -24,15 +24,15 @@
...
-		"@storybook/react": "8.0.0",
-		"@storybook/react-webpack5": "8.0.0",
-		"@storybook/theming": "8.0.0",
+		"@storybook/addon-a11y": "8.0.5",
+		"@storybook/addon-docs": "8.0.5",
+		"@storybook/addon-essentials": "8.0.5",
+		"@storybook/addon-links": "8.0.5",
+		"@storybook/addon-viewport": "8.0.5",
+		"@storybook/manager-api": "^8.0.5",
```
Comment (SiAdcock) [inline package.json:32]: The replacement for `@storybook/addons`

**Review by jamesmockett — APPROVED** [review]
Shame it didn't fix the contrast check issue, but good to replace the deprecated addons package 👍

**Pull request merged**